### PR TITLE
Upmerge 16.05.2024

### PR DESCRIPTION
**Pull request by RobertGalatNordic**
new SMF api has more arguments.

## CI parameters

```yaml
Github_actions:
  #(branch, hash, pull/XXX/head)
  NRF_revision: pull/15351/head

  # Do not change after creating PR
  Create_NRF_PR: false
Jenkins:
  test-sdk-sidewalk: sdk-nrf-pr-15351
```

## Description

https://github.com/nrfconnect/sdk-nrf/pull/15351

## Self review

- [x] There is no commented code.
- [x] There are no TODO/FIXME comments without associated issue ticket.
- [x] Commits are properly organized.
- [ ] Change has been tested.
- [ ] Tests were updated (if applicable).

### CODE REVIEW

File: samples/sid_end_device/child_image/mcuboot/prj.conf
Comment (ktaborowski) [inline samples/sid_end_device/child_image/mcuboot/prj.conf:1]: Just asking: is it intentional to remoce boot baner also form debug configurations?

File: samples/sid_end_device/sysbuild/hci_ipc/prj.conf
Comment (ktaborowski) [inline samples/sid_end_device/sysbuild/hci_ipc/prj.conf:1]: minor: this is not dts for nrf54, but config for nrf53 netcore (misleading commit message)

**Review by ktaborowski — APPROVED** [review]
Loogs good overall

File: samples/sid_end_device/child_image/mcuboot/prj.conf
Comment (RobertGalatNordic) [reply samples/sid_end_device/child_image/mcuboot/prj.conf:1]: Yes, the boot banner increases the size of the bootloader, and therefore it does not fit in our dedicated space, so this also affects the debug versions of the application (I consider debug in terms of the debug version of the application, not the whole package, but in the future I would like to move away from custom, optimized size of the bootloader partition in favor of the default - configuration that could be common across different projects)

File: samples/sid_end_device/sysbuild/hci_ipc/prj.conf
Comment (RobertGalatNordic) [reply samples/sid_end_device/sysbuild/hci_ipc/prj.conf:1]: yes this should be in previous commit, it had to slip when I added files to commits

**Pull request merged**